### PR TITLE
Replaces the deprecated rustc_serialize with data_encoding

### LINKED ISSUE #29: Drop rustc-serialize dependency
It has been deprecated: [announcement](https://users.rust-lang.org/t/deprecation-of-rustc-serialize/10485?u=dtolnay).

### PR DESCRIPTION
**Pull request by zmanian**
Fixes #29

### CODE REVIEW

**Comment by zmanian** [comment]
Not at all sure why Travis failed...

**Comment by heartsucker** [comment]
```
error[E0463]: can't find crate for `rustc_serialize`
 --> tests/tests.rs:1:1
  |
1 | extern crate rustc_serialize;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate
error: aborting due to previous error
error: Could not compile `tuf`.
```

File: Cargo.toml [hunk 1]
```diff
@@ -1,44 +1,53 @@
 [package]
-name = "tuf"
-version = "0.0.2"
-authors = [ "heartsucker <heartsucker@autistici.org>" ]
+authors = ["heartsucker <heartsucker@autistici.org>"]
```
Comment (heartsucker) [inline Cargo.toml]: I'm not sure what you did, but can you leave these fields in the same order as there were originally. I don't see any reason to change their order.

File: Cargo.toml [hunk 2]
```diff
@@ -1,44 +1,53 @@
...
 ring = "0.7"
-rustc-serialize = "0.3"
 serde = "0.9"
 serde_derive = "0.9"
 serde_json = "0.9"
-url = "1.4"
 untrusted = "0.3"
+url = "1.4"
+
+[dependencies.chrono]
```
Comment (heartsucker) [inline Cargo.toml]: Also can you leave this in the JSON-esque style and not TOML style? I find it easier to skim my dependencies that way.

File: src/lib.rs [hunk 3]
```diff
@@ -3,7 +3,7 @@ extern crate ring;
 extern crate itoa;
 #[macro_use]
 extern crate log;
-extern crate rustc_serialize;
+extern crate data_encoding;
```
Comment (heartsucker) [inline src/lib.rs]: Nit: Alphabetical order for crates.

File: src/metadata.rs [hunk 4]
```diff
@@ -3,7 +3,8 @@ use json;
 use ring;
 use ring::digest::{digest, SHA256};
 use ring::signature::{verify, ED25519};
-use rustc_serialize::hex::{FromHex, ToHex};
+// use rustc_serialize::hex::{FromHex, ToHex};
```
Comment (heartsucker) [inline src/metadata.rs]: Remove the comment.

File: Cargo.toml (hunk 1 above)
Comment (zmanian) [inline Cargo.toml]: It looks like `cargo add` did that... I should fix that in cargo add

**Pull request merged**